Sync 'staging' to 'main'

JMBeresford opened the pull request:
The 'stage' branch has been deprecated in favor of this new 'staging' branch due to messy divergences and conflicts in the former.

This changes syncs main such that github action config files now include the new branch in their workflows, and finishes reverting the AWS changes that unblock local dev.

JMBeresford committed:
fix: revert ws redirection (#17)

Previously we had altered the UI to send it's chat network requests through an endpoint that was exclusively available on the deployed server in AWS. This was done using a reverse-proxy, which of course is not available in the dev environment.

This reverts that change in anticipation of a more robust solution, and unblocks local dev.

JMBeresford committed:
chore: add staging branch to GH actions

JMBeresford committed:
docs: update readme to include new staging branch

JMBeresford committed:
chore(server): clean up module/dir structure (#20)

Previously we had various modules in our server crate under subdirectories. Due to an incomplete understanding of the Rust module system, we introduced these modules into our package via a named file declaration at the root level in addition to the module directory itself. This means for each module in our package, we had two entries at the root directory level.

This change introduces mod.rs files into each module directory as an idiomatic alternative to root-level file declarations.

Donovanphenry reviewed (dismissed):
I like the idea of structuring our server code better. Having everything under utils is not future-proof whatsoever. This will make the changes from [my pr](https://github.com/fun-software/Fun-Game/pull/21) somewhat tedious, but we'll have to pay this price eventually, and might as well start now.

JMBeresford committed:
feat(client): start game from lobby (#19)

Currently there is no way to actually enter a game.

This change adds the ability to start a game from the lobby. This will currently
bring you to a placeholder scene with minimal controls. This change also made minor semantic changes to indicate that the global wrapping 'Scene' element is actually a simple Canvas elem that wraps the 'scene' proper.

resolves FG-76

JMBeresford committed:
chore(Client): implemented useGameState react hook (#22)

Currently we have a WebRTCClient class that is written in vanilla
typescript. While this will work for imperative use-cases, ultimately
we will want a declarative method of getting server updates in an
idiomatic React fashion.

This change adds such functionality, and refactors some client logic
slightly.

resolves FG-80
